modules/living_spaces_menus/living_spaces_menus.module

```diff
@@ -5,16 +5,62 @@
  * Contains functions and hooks for living_spaces_menus.module.
  */
 
-use Drupal\user\UserInterface;
+use Drupal\Core\Template\Attribute;
+use Drupal\Core\Url;
 
 /**
- * Implements hook_ENTITY_TYPE_update().
+ * Implements hook_collapsible_menu_alter().
  */
-function living_spaces_menus_user_update(UserInterface $user) {
-  $field = 'joined_spaces';
-  if ($user->hasField($field)) {
-    if (!$user->get($field)->equals($user->original->get($field))) {
-      \Drupal::service('plugin.manager.menu.link')->rebuild();
+function living_spaces_menus_dom_bootstrap_collapsible_menu_alter(&$build) {
+  $spaces_menu_key = 'menu_link_content:436e7a5e-7eb0-4ba7-928c-b9064865f174';
+  $have_favorites = FALSE;
+  $connection = \Drupal::database();
+
+  $query = $connection->select('groups_field_data', 'groups_field_data');
```

Comment:
You can use \Drupal::entityTypeManager()->getStorage('group')->getQuery() instead.
This way both living_spaces_group.exclude_spaces and space templates will be already excluded and you won't need those query alters that repeat already existing functionality

Reply:
Done!